### PR TITLE
doc: update examples for context sensitivity

### PR DESCRIPTION
**Pull request by KevinEady**
Closes: https://github.com/nodejs/node-addon-api/issues/1011

### CODE REVIEW

File: doc/class_property_descriptor.md [hunk 1]
```diff
@@ -31,8 +30,9 @@ Napi::Object Example::Init(Napi::Env env, Napi::Object exports) {
         InstanceAccessor<&Example::GetValue>("readOnlyProp")
     });
 
-    constructor = Napi::Persistent(func);
-    constructor.SuppressDestruct();
+    Napi::FunctionReference *constructor = new Napi::FunctionReference();
+    *constructor = Napi::Persistent(func);
+    env.SetInstanceData(constructor);
```
Comment (legendecas) [inline doc/class_property_descriptor.md:35]: A finalizer is required for this FunctionReference pointer right?

Comment (KevinEady) [reply doc/class_property_descriptor.md:35]: Hi @legendecas ,

Good question... I was under the impression that the instance data would be `delete`d, so there would be no need for a Finalizer.
- In the case of multiple calls to `Example::Init`, the previous instance data would be overwritten via most recent call to `env.SetInstanceData`. https://github.com/nodejs/node/blob/d615aeb7583b15bb5a8d1ec666ea29b8c7377455/src/js_native_api_v8.cc#L3201-L3205 does this delete the data?
- The final data would be `delete`d on environment cleanup.

Is my understanding correct?

Comment (legendecas) [reply doc/class_property_descriptor.md:35]: Eh, IIUC, it's node-addon-api's default finalizer deleted the data pointer: https://github.com/nodejs/node-addon-api/blob/main/napi.h#L208. Not quite intuitive tho... So no finalizer is required here.

Comment (legendecas) [reply doc/class_property_descriptor.md:35]: The `Delete` you point out only deleting the `RefBase` structure allocated in Node.js core, which would invoke the finalizer provided by node-addon-api.

**Review by legendecas — APPROVED** [review]

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
Landed as 627dbf3c37ab64e33ad4431c6b569f6a36dd1723

**Pull request closed without merging**